### PR TITLE
Chore: clean code

### PR DESCRIPTION
**Pull request by savion1024**
Improve code readability.

### CODE REVIEW

File: component/resolver/resolver.go [hunk 1]
```diff
@@ -141,12 +141,10 @@ func LookupIPWithResolver(ctx context.Context, host string, r Resolver) ([]net.I
 		return []net.IP{node.Data.(net.IP)}, nil
 	}
 
-	if r != nil {
-		if DisableIPv6 {
-			return r.LookupIPv4(ctx, host)
```
Comment (Dreamacro) [inline component/resolver/resolver.go:146]: This is not equivalent, in that `r != nil`, you need to use `r` itself

Comment (savion1024) [reply component/resolver/resolver.go:146]: thanks for u reply

**Comment by savion1024** [comment]
Hello,  i think this change may can Improve code readability.Please take a look. thx~

**Comment by Dreamacro** [comment]
I don't think it improves the readability of the code, can you explain? Also the error code from the previous commit is still there.

**Comment by savion1024** [comment]
the logic of both `if` and `else` blocks is to return, it can be replaced with multiple `if` statements, which, in my understanding, can make the code more readable, especially within the `LookupIPWithResolver` method. If you agree with my point of view, I will submit another pull request. This PR will only have one commit. If it's not acceptable, I will close the PR. Thank you for your response

**Pull request closed without merging**